convert2rhel/actions/report.py

```diff
@@ -0,0 +1,155 @@
+# Copyright(C) 2023 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__metaclass__ = type
+
+import logging
+
+from convert2rhel.actions import STATUS_CODE
+
+
+logger = logging.getLogger(__name__)
+
+
+def _dictionary_value_lookup(dictionary=None, look_for=None):
+    """Return the key associated with the value.
+
+    .. important:: This function will not do a depth search for the dictionary,
+        it is meant to be used with dictionaries of one level.
+
+    :param dictionary: The dictionary to be used in the revese search.
+    :type dictionary: Mapping
+    :param look_for: The value to be used in the revese search.
+    :type look_for: Any
+
+    :return: The expected key inside the dictionary
+    :rtype: str | None
+    """
+    for key, value in dictionary.items():
+        if value == look_for:
+            return key
+
```

Comment:
Since STATUS_CODE is constant and we're going to be looking up the same status multiple times, I think I would implement this by switching keys and values once and then consulting that dictionary for the proper status name.  (Note: In python3, another way to do this would be to use `@functools.lru_cache`)

```suggestion
_STATUS_NAME_FROM_CODE = dict((value, key) for key, value in actions.STATUS_CODE.items())
```